packages/vesting-base-lp/src/handlers.rs

```diff
@@ -0,0 +1,819 @@
+            deps.storage,
+            env.block.height,
+            |s| {
+                let mut state = s.ok_or(ContractError::AmountIsNotAvailable {})?;
+                state.total_released = state.total_released.checked_add(claim_amount)?;
+                Ok(state)
+            },
+        )?;
+    };
+
+    Ok(response.add_attributes(vec![
+        attr("action", "claim"),
+        attr("address", &info.sender),
+        attr("available_amount", available_amount),
+        attr("claimed_amount", claim_amount),
+    ]))
+}
+
+pub(crate) fn set_vesting_token(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    token: AssetInfo,
+) -> Result<Response, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+    if info.sender != config.owner && info.sender != config.token_info_manager {
+        return Err(ContractError::Unauthorized {});
+    }
+    token.check(deps.api)?;
+    config.vesting_token = Some(token);
+
+    CONFIG.save(deps.storage, &config)?;
+    Ok(Response::new())
+}
+
+pub(crate) fn get_vesting_token(config: &Config) -> Result<AssetInfo, ContractError> {
+    config
+        .vesting_token
+        .clone()
+        .ok_or(ContractError::VestingTokenIsNotSet {})
+}
+
+fn execute_migrate_liquidity(
+    deps: DepsMut,
+    env: Env,
+    slippage_tolerance: Option<Decimal>,
+) -> Result<Response, ContractError> {
+    let migration_state: MigrationState = MIGRATION_STATUS.load(deps.storage)?;
+    if migration_state == MigrationState::Completed {
+        return Err(ContractError::MigrationComplete {});
+    }
+    let migration_config: XykToClMigrationConfig = XYK_TO_CL_MIGRATION_CONFIG.load(deps.storage)?;
+
+    let vesting_infos = read_vesting_infos(
+        deps.as_ref(),
+        migration_config.last_processed_user,
+        Some(migration_config.batch_size),
+        None,
+    )?;
+
+    let vesting_accounts: Vec<_> = vesting_infos
+        .into_iter()
+        .map(|(address, info)| VestingAccountResponse { address, info })
+        .collect();
+
+    if vesting_accounts.is_empty() {
+        MIGRATION_STATUS.save(deps.storage, &MigrationState::Completed)?;
+    }
+    let mut resp = Response::default();
+
+    // get pairs LP token addresses
+    let pair_info: PairInfo = deps
+        .querier
+        .query_wasm_smart(migration_config.xyk_pair.clone(), &PairQueryMsg::Pair {})?;
+
+    // query max available amounts to be withdrawn from pool
+    let max_available_amount = {
+        let resp: BalanceResponse = deps.querier.query_wasm_smart(
+            pair_info.liquidity_token.clone(),
+            &Cw20QueryMsg::Balance {
+                address: env.contract.address.to_string(),
+            },
+        )?;
+        resp.balance
+    };
+
+    for user in vesting_accounts.into_iter() {
+        if max_available_amount.is_zero() {
+            return Ok(resp);
+        }
+
+        let user_amount = compute_share(&user.info)?;
```

Comment:
if `user_amount==0` we could just skip all the callback chain for the user, 0 old_lp in any case translates in 0 new_lp